packages/ui/src/components/Button/Button.tsx

```diff
@@ -19,21 +19,20 @@ export function Button({
     rounded,
     ...buttonProps
 }: Button.Props) {
-    const iconClassName = twMerge(
-        !children && variation !== 'navigation'
-            ? // Adjusted icon height for square buttons
-              'w-4 h-6'
-            : 'w-6 h-6',
-    );
+    const isIconOnly = Boolean(icon && !children);
+    const hasLeftIcon = Boolean(icon && iconPlacement === 'left');
+    const hasRightIcon = Boolean(icon && iconPlacement === 'right');
 
     return (
         <button
```

Comment:
Nice one @MohammadxAli - wondering if just a `w-*` would suffice? Taking a look now at the PR.

Reply:
They're already using a `w-*` class :)

![image](https://github.com/prezly/theme-kit-js/assets/67554982/6145e509-c7a0-44b0-b40d-d800067d5485)

Another option is to utilize [`shrink-0`](https://tailwindcss.com/docs/flex-shrink) (`flex-shrink: 0`) on the icon to prevent it from shrinking. This way, a `min-w-*` is no longer necessary.

Reply:
Updated